test/fun_with_flags/notifications/redis_test.exs

```diff
@@ -7,6 +7,91 @@ defmodule FunWithFlags.Notifications.RedisTest do
 
   @moduletag :redis_pubsub
 
+  describe "worker_spec" do
+    setup do
+      # Before each test ensure that the initial config is the default one.
+      ensure_default_redis_config_in_app_env()
+
+      # Cleanup
+      on_exit(&reset_app_env_to_default_redis_config/0)
+      :ok
+    end
+
+    test "when the Redis config is a URL string" do
+      url = "redis:://1.2.3.4:5678/42"
+      configure_redis_with(url)
+
+      expected = %{
+        id: FunWithFlags.Notifications.Redis,
+        start: {
+          FunWithFlags.Notifications.Redis,
+          :start_link,
+          [
+            {url, name: :fun_with_flags_notifications, sync_connect: false}
+          ]
+        },
+        type: :worker,
+        restart: :permanent
+      }
+
+      assert ^expected = NotifiRedis.worker_spec()
+    end
+
+    test "when the Redis config is a {URL, opts} tuple" do
+      url = "redis:://1.2.3.4:5678/42"
+      opts = [socket_opts: [:inet6]]
+      configure_redis_with({url, opts})
+
+      expected = %{
+        id: FunWithFlags.Notifications.Redis,
+        start: {
+          FunWithFlags.Notifications.Redis,
+          :start_link,
+          [
+            {
+              url,
+              [
+                socket_opts: [:inet6],
+                name: :fun_with_flags_notifications,
+                sync_connect: false
+              ]
+            }
```

Comment:
❓ Is this extra wrapping as a tuple OK? The shape of this is slightly different than Persistance.Redis.worker_spec/0. Maybe that's OK since that function is based on Redix.child_spec/1?

Reply:
Yes, that's right. It just happens that `Redix` wants its arguments in a way and this FWF's supervised worker wants its arguments in a different way.

Here the format is:
```
start: {
  Module.That.Receives.The.Function.Call,
  :the_name_of_the_function,
  [
    "a list that contains"
    "the",
    {"positional", "arguments", "to", "the function"},
    :they,
    [
      could: "be",
      anything: "really"
    ]
  ]
}
```

The name of the function and the arguments inside the outermost `[]` are application specific. It's just customary to use `start_link` because that's the convention.

For example FWF still works with this diff:

```diff
diff --git a/lib/fun_with_flags/notifications/redis.ex b/lib/fun_with_flags/notifications/redis.ex
index 3f4c54a..aee4930 100644
--- a/lib/fun_with_flags/notifications/redis.ex
+++ b/lib/fun_with_flags/notifications/redis.ex
@@ -27,7 +27,7 @@ defmodule FunWithFlags.Notifications.Redis do

     %{
       id: __MODULE__,
-      start: {__MODULE__, :start_link, [redis_conn_config]},
+      start: {__MODULE__, :foobar, [:something_else, redis_conn_config]},
       restart: :permanent,
       type: :worker,
     }
@@ -39,7 +39,7 @@ defmodule FunWithFlags.Notifications.Redis do
   # used to build the outgoing notification payloads and to ignore
   # incoming messages that originated from this node.
   #
-  def start_link(redis_conn_config) do
+  def foobar(:something_else, redis_conn_config) do
     GenServer.start_link(__MODULE__, {redis_conn_config, Config.build_unique_id}, [name: __MODULE__])
   end
```